ts: fix query sample size

jasonlmfong opened the pull request:
currently the sample duration is always defaulted to 10s, even when supplied in query parameter

Epic: None

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/156589)
<!-- Reviewable:end -->

dhartunian commented on pkg/ts/server.go:
```diff
@@ -339,7 +348,7 @@ func (s *Server) Query(
 						query,
-						Resolution10s,
+						diskResolution,
 						timespan,
```
if timespan is passed in, and contains `sampleNanos`, and if the `resolution` is derived from `sampleNanos`, why should we pass it in at all? Can't the implementation of `db.Query` just do it?

jasonlmfong replied:
I decided to keep the query.go logic as-is, instead only modifying the server.go disk resolution based on the roll up time. this way the client gets what they asked for, without too much magic happening.

jasonlmfong committed:
ts: fix ts/query disk resolution size

currently the disk resolution is always defaulted to 10s, even when the data has been rolledup

Epic: None
Release note: None
